WordPress/Classes/Models/Blog+Post.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+public extension Blog {
+    /// Lookup a post in the blog.
+    ///
+    /// - Parameter postID: The ID associated with the post.
+    /// - Returns: The `AbstractPost` associated with the given post ID.
+    @objc(lookupPostWithID:inContext:)
+    func lookupPost(withID postID: NSNumber, in context: NSManagedObjectContext) -> AbstractPost? {
+        lookupPost(withID: postID.int64Value, in: context)
+    }
+
+    /// Lookup a post in the blog.
+    ///
+    /// - Parameter postID: The ID associated with the post.
+    /// - Returns: The `AbstractPost` associated with the given post ID.
+    func lookupPost(withID postID: Int64, in context: NSManagedObjectContext) -> AbstractPost? {
+        let request = NSFetchRequest<AbstractPost>(entityName: NSStringFromClass(AbstractPost.self))
+        request.predicate = NSPredicate(format: "blog = %@ AND original = NULL AND postID = %@", self, NSNumber(value: postID))
+        return (try? context.fetch(request))?.first
+    }
+}
```

Comment:
I've been thinking about this style of methods, which other recent PRs in this Core Data improvements series have implemented.

The thing I've been wandering is whether it would be best to make these `NSManagedObjectContext` extensions.

I think this approach its best.

Extending `NSManagedObjectContext` while functionally correct, would put the functionality at the wrong abstraction level. That is, we are dealing with model entities and their relationships at this level, but to me `NSManagedObjectContext` seems closer to the raw storage implementation detail.

This implementation reminds me a bit of [Active Record](https://www.martinfowler.com/eaaCatalog/activeRecord.html).